Feature/ASSA-10 Additional detekt configurations for Compose and code cleanup

Josh-Gebbeken opened the pull request:
Added additional Detekt configuration for Compose:

- Magic Numbers for Compose like 32.dp 
- PreviewAllDevices annotation
- File-wide suppression for ther Dimens class like ConstructorParameter, MatchingDeclarationName, LongParameterList
- Additional work to break down long methods
- Refactored files that don't have a matching class name to match. Refactored based on actual name of screen.
- Fixed additional UnusedPrivateMembers

JGebbeken-BR committed:
ASAA-10 Added Additional Detekt compose rule exceptions

Josh-Gebbeken committed:
feature/ASAA-10 fixed Detekt code smell issues

Josh-Gebbeken committed:
feature/ASAA-10 Fixed additional code smells

Josh-Gebbeken committed:
Merge branch 'compose/develop' into feature/ASAA-10

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt:
```diff
@@ -12,12 +12,12 @@ object Routes {
     const val Profile = "profile"
 
-    fun RepositoryBrowser(data: RepositoryBrowserData) =
+    fun repositoryBrowser(data: RepositoryBrowserData) =
```
@Josh-Gebbeken @BottleRocket-Colin I'm honestly on the fence about this one. If we had used sealed classes (which we have already discussed and decided against as they don't bring a lot to the table) then all of the sub objects and data classes would be UpperCamelCased.

Since we're using an Object, detekt is going to complain when we use UpperCamelCase function names that take in additional route data to mimic the casing for sealed classes/other Composable functions.

Thoughts?

jaredstockton replied:
@BottleRocket-Colin Any input on this one?

BottleRocket-Colin replied:
Since it's a noun and it maintains consistency with the rest of  file, I'd say we should add a suppress notation to Routes for this one and keep it PascalCase.

Josh-Gebbeken replied:
Changed back to PascalCase and added the suppression

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt:
```diff
@@ -48,38 +48,13 @@ import com.bottlerocketstudios.launchpad.compose.bold
             Column(
                 modifier = Modifier
-                    .fillMaxSize()
+                    .fillMaxSize().padding(paddingValues)
```
@Josh-Gebbeken Was this addition intentional?

BottleRocket-Colin replied:
@br-jared-stockton - This is following the convention defined here: https://developer.android.com/jetpack/compose/layouts/material#scaffold

Good enhancement @Josh-Gebbeken 

I was being sloppy by adding the suppress notation.

Josh-Gebbeken replied:
It was highlighting in red for me so I ended up adding it to remove.

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:
```diff
@@ -27,11 +27,6 @@ import com.bottlerocketstudios.compose.util.convertToImageBitmap
 import com.bottlerocketstudios.launchpad.compose.light
```
@Josh-Gebbeken Should this file be refactored to `FileBrowserScreen`?

BottleRocket-Colin replied:
Couldn't hurt.

Josh-Gebbeken replied:
yeah that should work

Josh-Gebbeken replied:
This has changed to `FileBrowserScreen`

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt:
```diff
@@ -188,3 +180,12 @@ private fun PreviewOuterScreenContent() {
+    val onRepositoryItemClicked: (RepositoryItemUiModel) -> Unit
+)
+
+private const val DEFAULT_ITEM_COUNT = 3
```
@Josh-Gebbeken This provides more visibility when pulled out 👍🏻

BottleRocket-Colin replied:
@Josh-Gebbeken - Since this value needs to stay in sync with the list<RepositoryItemUiModel> it may be better to val that list a constant and then get the size of the list.  This would avoid maintaining this constant.

jaredstockton commented on compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt:
```diff
@@ -19,6 +19,7 @@ fun ProvideColors(
 private val LocalAppColors = staticCompositionLocalOf {
     lightColors
 }
+private const val SMALL_SCREEN_WIDTH = 360
```
@Josh-Gebbeken nit: Refactor name to `SMALL_SCREEN_WIDTH_DP` for clarity on the unit of the value. If it stays as is, you would need to reference its usage to determine if the value represented dps or pixels :)

Josh-Gebbeken replied:
ok

jaredstockton commented on domain/src/main/java/com/bottlerocketstudios/brarchitecture/domain/models/Status.kt:
```diff
@@ -96,4 +96,5 @@ suspend fun <T : Any> logWrappedExceptions(className: String, methodName: String
 
 // Simple version of wrap exceptions; with no logging.
 suspend fun <T : Any> wrapExceptions(block: suspend () -> Status<T>): Status<T> =
+    @Suppress("TooGenericExceptionCaught") // this might be overly broad (catching NPEs isn't ideal) but not worth it yet to check for all other possible types (IO..., Json..., others)
```
@Josh-Gebbeken nit: Refactor comment to use `catching general Exceptions isn't ideal` instead of `catching NPEs isn't ideal`

Josh-Gebbeken replied:
That was a copy pasta of another comment on this file. I can change that as well as this for better clarification.

jaredstockton commented on config/detekt/detekt.yml:
```diff
@@ -681,7 +680,7 @@ style:
       - '2'
     ignoreHashCodeFunction: true
-    ignorePropertyDeclaration: false
+    ignorePropertyDeclaration: true
```
@Josh-Gebbeken I'm just curious where you see instances in the codebase where this rule is broken that caused you to change the value to true. Can you share a few?

BottleRocket-Colin replied:
@br-jared-stockton - I think this is for the colors file ?  Is that right @Josh-Gebbeken ?

Josh-Gebbeken replied:
Yes for the magic numbers in the compose color for example val black = Color(0xff000000)

jaredstockton replied:
Looks like this is one of the compose related solutions mentioned in the detekt docs: https://detekt.dev/docs/introduction/compose/#magicnumber

This should be fine to keep as is :)

jaredstockton reviewed (changes requested):
@Josh-Gebbeken Thanks for the PR! I've left some comments of things to address prior to approval :)

BottleRocket-Colin commented on compose/src/main/java/com/bottlerocketstudios/compose/appbar/ArchAppBar.kt:
```diff
@@ -62,3 +58,9 @@ fun ArchAppBar(
         )
     }
 }
+
```
Curious as to why this was moved below the composable function.  This makes it slightly harder to find.  Potentially much harder, if the file has a lot of composable functions after the main one.

Josh-Gebbeken replied:
This was one of the MatchingDeclarationName issues listed on Detekt. I thought It might of been easier to move it to the bottom and so this function can satisfy the issue.

Josh-Gebbeken replied:
Discussed over standup. We will move the Different Data Class states and put it in its own file in the same package

BottleRocket-Colin commented on compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt:
```diff
@@ -66,43 +66,8 @@ fun CardLayout(userRepositoryUiModel: UserRepositoryUiModel, selectItem: (userRe
-                            .fillMaxWidth()
-                    )
-                }
+               RepoDescriptionComponent(userRepositoryUiModel = userRepositoryUiModel, Modifier.weight(2f) )
```
In an effort to make this a bit more generic, lets change function signature of RepoDescriptionContent to accept each value: repo.name, display name, and description.  That way it's not coupled to UI model.

Josh-Gebbeken replied:
Sounds like a good idea.

Josh-Gebbeken replied:
Refactored out UI model so the component is decoupled.

BottleRocket-Colin commented on compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt:
```diff
@@ -31,50 +30,14 @@ import com.bottlerocketstudios.compose.util.PreviewAllDevices
-                )
-                .wrapContentSize()
-        )
+        ProfileDetailComponent(state = state)
```
Same as an earlier comment, let's pull out just what is needed from state and pass that along to ProfileDetailComponent.

Josh-Gebbeken replied:
ok

Josh-Gebbeken replied:
Moved to its own file within the same package

BottleRocket-Colin reviewed (changes requested):
A couple of minor nits in my comments. Only major issue I have is the moving of State objects below the screen in compose module.  Curious as to what error Detekt was throwing and if it could be suppressed for that file.  I did that b/c there's a lot of back and forth scrolling and it felt faster to have the state at the top of file near the top of the screen composable.

Josh-Gebbeken committed:
ASAA-10 Moved data classes that were states to individual files in the same package

ASAA-10 refactored out UI models from other compose components so that they are decoupled

Josh-Gebbeken committed:
ASAA-10 renaming constant to better suit what it is

Josh-Gebbeken committed:
ASAA-10 Ktlint formatting and checks

Josh-Gebbeken committed:
ASAA-10 refactored list and removed const val

Josh-Gebbeken committed:
ASAA-10 suppressed function name on Routes and reverted function names to be PascalCase

jaredstockton commented:
> A couple of minor nits in my comments. Only major issue I have is the moving of State objects below the screen in compose module. Curious as to what error Detekt was throwing and if it could be suppressed for that file. I did that b/c there's a lot of back and forth scrolling and it felt faster to have the state at the top of file near the top of the screen composable.

@Josh-Gebbeken Will detekt complain if we keep the State objects at the top of the composable screen files? I agree with @BottleRocket-Colin that it would be nicer to see the contract right at the top and then have the actual screen composable implementation below :)

Josh-Gebbeken commented:
> > A couple of minor nits in my comments. Only major issue I have is the moving of State objects below the screen in compose module. Curious as to what error Detekt was throwing and if it could be suppressed for that file. I did that b/c there's a lot of back and forth scrolling and it felt faster to have the state at the top of file near the top of the screen composable.
> 
> @Josh-Gebbeken Will detekt complain if we keep the State objects at the top of the composable screen files? I agree with @BottleRocket-Colin that it would be nicer to see the contract right at the top and then have the actual screen composable implementation below :)

Detekt will complain it being on top. Colin and I ended up agreeing to just put them in a separate file in the same package
